### ISSUE TITLE
Deep copying OrderedDicts

### ISSUE THREAD

**Issue by pcmoritz** [JuliaLang/DataStructures.jl#75]
Dear all,

there might be an issue in how OrderedDicts are handled by deepcopy:

```julia
julia> d1 = OrderedDict([("hi", 1)])
OrderedDict{ASCIIString,Int64} with 1 entry:
  "hi" => 1

julia> d2 = deepcopy(d1)
OrderedDict{ASCIIString,Int64} with 1 entry:
  "hi" => 1

julia> d2["hi"] = 2
2

julia> d1
OrderedDict{ASCIIString,Int64} with 1 entry:
  "hi" => 2

I would expect a behaviour similar as this one:

julia> d1 = Dict([("hi", 1)])
Dict{ASCIIString,Int64} with 1 entry:
  "hi" => 1

julia> d2 = deepcopy(d1)
Dict{ASCIIString,Int64} with 1 entry:
  "hi" => 1

julia> d2["hi"] = 2
2

julia> d1
Dict{ASCIIString,Int64} with 1 entry:
  "hi" => 1
```

I am using Julia 0.3.5 and DataStructures 0.3.5.

Let me know if you have an idea where the problem is or if you have an idea for a workaround. Thanks in advance!

Best,
Philipp.

*Edit: Added formatting (kms)*

**Comment by kmsquire** [comment]
Interestingly, this works on v0.4.

**Comment by kmsquire** [comment]
This was fixed on Julia 0.4 by #8560, but that was never backported to Julia v0.3.  That should happen (I believe) with the next point release of v0.3.

**Comment by pcmoritz** [comment]
Thank you!
